src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -584,6 +584,20 @@ private void CompileNamedType(NamedTypeSymbol containingType)
                 }
             }
 
+            foreach (var member in members)
+            {
+                if (member is SourcePropertySymbolBase property)
+                {
+                    var getMethod = property.GetMethod;
+                    var setMethod = property.SetMethod;
+                    if ((getMethod is null || PassesFilter(_filterOpt, getMethod)) &&
```

Comment:
>(getMethod is null || PassesFilter(_filterOpt, getMethod) [](http://example.com/codeflow?start=24&length=57)

There is a slight optimization that comes to mind. We can avoid checking accessors that failed syntactic check for a ```field``` keyword. That will require more tests, so a PROTOTYPE comment is good for now. The current behavior is correct (it resets the backing field only when it is safe to do so), but the implementation can be more "optimal".